packages/design-system/src/components/web-components/preactement/define.ts

```diff
@@ -359,9 +368,30 @@ function renderPreactComponent(this: CustomElement) {
   // Putting the original inner content into a template also allows us to keep a copy of
   // it for future renders where context has been lost (see function documentation).
   let template: HTMLTemplateElement | undefined = [...this.childNodes].find(isTemplate);
+  if (template && isEmptyTemplate(template)) {
+    // Web components rendered with Angular will have no innerHTML content at first, even
+    // if content was placed between the tags in the Angular template. In that case when
+    // we do our first render pass executing this function, we will generate an empty
+    // internal template because it gets filled with non-existent innerHTML. So when we
+    // perform a subsequent render, if our previous template is empty, we want to both
+    // start over with a new template and remove the old one so it doesn't make its way
+    // into the new. Even if the empty template is a false positive for this Angular
+    // behavior, there's no harm in replacing it with a new empty template, but there
+    // _is_ harm in leaving a non-empty template to duplicate its content by using it in
+    // the inner HTML that will go into a new template (which results in buttons inside
+    // of buttons and things like that).
+    template.remove();
+    template = undefined;
+  }
   if (!template) {
     template = document.createElement('template');
-    template.innerHTML = wrapTemplateHtml(this.innerHTML);
+    if (addedNodes) {
+      const span = document.createElement('span');
+      span.append(...addedNodes);
```

Comment:
Nit: Is there a reason you chose a span over a div? I'm not sure there's a difference in how inline vs. block content is handled by the template.

Reply:
The template wrapping functions also use a span, and I originally chose span because not all web components are block components, and a span wouldn't mess with that.